packages/circuit-ui/components/SideNavigation/components/PrimaryLink/PrimaryLink.tsx

```diff
@@ -0,0 +1,255 @@
+import isPropValid from '@emotion/is-prop-valid';
+import { Theme } from '@sumup/design-tokens';
+import { ArrowRight } from '@sumup/icons';
+
+import styled, { StyleProps } from '../../../../styles/styled';
+import {
+  focusVisible,
+  disableVisually,
+  hideVisually,
+  cx,
+} from '../../../../styles/style-mixins';
+import { useClickEvent } from '../../../../hooks/useClickEvent';
+import { useComponents } from '../../../ComponentsContext';
+import Body from '../../../Body';
+import { PrimaryLinkProps as PrimaryLinkType } from '../../types';
+import { ClickEvent } from '../../../../types/events';
+import { uniqueId } from '../../../..';
+
+export interface PrimaryLinkProps extends PrimaryLinkType {
+  isOpen?: boolean;
+  suffix?: FC<{ className?: string; role?: string }>;
+}
+
+type AnchorProps = Pick<PrimaryLinkProps, 'isActive' | 'isOpen'> & {
+  hasBadge: boolean;
+};
+
+const anchorStyles = ({ theme }: StyleProps) => css`
+  position: relative;
+  display: flex;
+  align-items: center;
+  width: 100%;
+  background: none;
+  border: none;
+  outline: none;
+  text-align: left;
+  cursor: pointer;
+  text-decoration: none;
+  color: ${theme.colors.black};
+  padding: ${theme.spacings.kilo} ${theme.spacings.giga};
+  transition: color ${theme.transitions.default},
+    background-color ${theme.transitions.default};
+
+  &:hover {
+    background-color: ${theme.colors.n100};
+  }
+
+  &:active {
+    background-color: ${theme.colors.n200};
+  }
+
+  &:disabled {
+    ${disableVisually()};
+  }
+
+  ${theme.mq.untilGiga} {
+    /* Prevent the border from being obscured by the sub navigation  */
+    margin-bottom: 1px;
+
+    &::after {
+      content: '';
+      display: block;
+      position: absolute;
+      top: 100%;
+      right: ${theme.spacings.giga};
+      left: ${theme.spacings.giga};
+      width: calc(100% - 2 * ${theme.spacings.giga});
+      border-bottom: ${theme.borderWidth.kilo} solid ${theme.colors.n200};
+      transition: width ${theme.transitions.default},
+        right ${theme.transitions.default}, left ${theme.transitions.default};
+    }
+  }
+
+  ${theme.mq.giga} {
+    padding: ${theme.spacings.kilo};
+    margin-bottom: ${theme.spacings.kilo};
+  }
+`;
+
+const anchorActiveStyles = ({ theme, isActive }: StyleProps & AnchorProps) =>
+  isActive &&
+  css`
+    svg {
+      color: ${theme.colors.p500};
+    }
+
+    ${theme.mq.giga} {
+      color: ${theme.colors.p500};
+
+      &:hover {
+        background-color: ${theme.colors.p100};
+      }
+    }
+  `;
+
+const anchorOpenStyles = ({ theme, isOpen }: StyleProps & AnchorProps) =>
+  isOpen &&
+  css`
+    ${theme.mq.untilGiga} {
+      &::after {
+        right: 0;
+        left: 0;
+        width: 100%;
+      }
+    }
+  `;
+
+const anchorBadgeStyles = ({ theme, hasBadge }: StyleProps & AnchorProps) =>
+  hasBadge &&
+  css`
+    &::before {
+      display: block;
+      content: '';
+      position: absolute;
+      top: 14px;
+      left: 47px;
```

Comment:
Lots of magic numbers here, is there a way to position it differently? For example relative to the icon rather than the anchor?

Reply:
Not really, no. It would have to be inside the icon, i.e. inside the SVG, to be relative to it. I'll add some comments/formulas here.

Reply:
In the dashboard I had an icon wrapper and that's where I set the badge ([source](https://github.com/sumup/ze-dashboard/blob/895aff15aacb481cad94dbf9072a2f422b6f0e2e/src/components/SideNav/components/NavItem/NavItem.js)). The downside is more DOM nesting, benefit is a cleaner positioning